OpenRA.Game/Game.cs

```diff
@@ -401,6 +402,35 @@ static string ChooseShellmap()
 		public static void RunAfterTick(Action a) { delayedActions.Add(a); }
 		public static void RunAfterDelay(int delay, Action a) { delayedActions.Add(a, delay); }
 
+		static void TakeScreenshotInner()
+		{
+			Log.Write("debug", "Taking screenshot");
+
+			Bitmap bitmap;
+			using (new PerfTimer("Renderer.TakeScreenshot"))
+				bitmap = Renderer.TakeScreenshot();
+
+			ThreadPool.QueueUserWorkItem((o) =>
+			{
+				var mod = ModData.Manifest.Mod;
+				var directory = Platform.ResolvePath("^", "Screenshots", mod.Id, mod.Version);
+				Directory.CreateDirectory(directory);
+
+				var format = Settings.Graphics.ScreenshotFormat;
+				var extension = ImageCodecInfo.GetImageEncoders().FirstOrDefault(x => x.FormatID == format.Guid)
+					.FilenameExtension.Split(';').First().ToLowerInvariant().Substring(1);
+				var filename = TimestampedFilename();
+				var destination = Path.Combine(directory, string.Concat(filename, extension));
+				var stream = File.Create(destination);
```

Comment:
Bitmap has a `Save()` method.  Why the extra overhead of creating and writing to a stream?